### ISSUE TITLE
bam_io error

### ISSUE THREAD

**Issue by ShujiaHuang** [seqan/seqan#908]
Hi! I was trying the simple example of [SamBamIO](http://seqan.readthedocs.org/en/master/Tutorial/SamBamIO.html) in seqan tutorial . I do the same things as it's said in the tutorial. Then I just directly copy the [example](http://seqan.readthedocs.org/en/master/Tutorial/SamBamIO.html#a-first-working-example) to test the bam_io. it can success in compiling process, but when I try to execute the program , I got the following error : 

```
@HD
@SQ
@SQ
r001	163	../../include/seqan/sequence/string_base.h:439 Assertion failed : static_cast<TStringPos>(pos) < static_cast<TStringPos>(length(me)) was: 1 >= 0 (Trying to access an element behind the last one!)
Abort trap: 6
```

and the input samfile that I use is also in the tutorial:

```
@HD VN:1.3  SO:coordinate
@SQ SN:ref  LN:45
@SQ SN:ref2 LN:40
r001	163	ref	7	30	8M4I4M1D3M	=	37	39	TTAGATAAAGAGGATACTG	*	XX:B:S,12561,2,20,112
r002	0	ref	9	30	1S2I6M1P1I1P1I4M2I	*	0	0	AAAAGATAAGGGATAAA	*
r003	0	ref	9	30	5H6M	*	0	0	AGCTAA	*
r004	0	ref	16	30	6M14N1I5M	*	0	0	ATAGCTCTCAGC	*
r003	16	ref	29	30	6H5M	*	0	0	TAGGC	*
r001	83	ref	37	30	9M	=	7	-39	CAGCGCCAT	*
```

 What can be the problem here?  (I use the latest seqan version in github) Thank you!

**Comment by xp3i4** [comment]
Hi,  you can try to replace the sentence 
```c++
BamFileOut bamFileOut( std::cout, Sam()); 
```
int the demos/tutorial/bam_io/example1.cpp to the following sentence.
```c++
BamFileOut bamFileOut(context(bamFileIn), std::cout, Sam());
```

This might work in your case.

**Comment by esiragusa** [comment]
Fixed by #916 / #917.
